### PR TITLE
PTP capability

### LINKED ISSUE #440: Support for PTP capabilities
**Issue Type**

- Template enhancement
- Role enhancement

**Is your feature request related to a problem? Please describe.**
The idea is to support PTP knobs on interface level and global mode.
This will allow the devices in the DC to speak PTP.

Later could be enhanced for documentation.

**Describe the solution you'd like**
The following as an example of knobs required.

```
interface Ethernet {{ INTERFACE_NUMBER }}
   ptp enable
!
ptp clock-identity {{ PTP_CLOCK_ID }}
ptp source ip {{ PTP_SOURCE_IP }}
ptp message-type general dscp {{ VALUE }} default
ptp message-type event dscp {{ VALUE }} default
ptp priority1 {{ PTP_PRIORITY_1 }}
ptp priority2 {{ PTP_PRIORITY_2 }}
ptp ttl {{ PTP_TTL }}
```

**Describe alternatives you've considered**
Static configuration or custom template

**Additional context**

### PR DESCRIPTION
**Pull request by onurgashi**
## Change Summary

Allows implementation of PTP in the eos_cli_config_gen

Next step add more knobs and maybe create data model for boradcast & media.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #440
 
## Component(s) name
eos_cli_config_gen

## Proposed changes

Interface level and global ptp.j2

## How to test

Successful venv/physical DUT, and molecule tests.

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
Please rebase this PR since project branch was rebased before it merged into devel.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{# eos - PTP #}
+{% if ptp is defined and ptp is not none %}
+!
+{%   if ptp.clock_identity is defined and ptp.clock_identity is not none %}
+ptp clock-identity {{ ptp.clock_identity }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2:5]: Indention is only 2 here. Should be 4.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2 [hunk 2]
```diff
@@ -0,0 +1,42 @@
+### PTP Summary
+{% if ptp is defined and ptp is not none %}
+| PTP setting | Value |
+| ----------- | ----- |
+{%   if ptp.clock_identity is defined and ptp.clock_identity is not none %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2]: indention only 2. Should be 4.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+| Priority2 | {{ ptp.priority2 }} |
+{%   endif %}
+{%   if ptp.ttl is defined and ptp.ttl is not none %}
+| TTL | {{ ptp.ttl }} |
+{%   endif %}
+{%   if ptp.message_type is defined and ptp.message_type is not none %}
+{%       if ptp.message_type.general is defined and ptp.message_type.general is not none %}
+{%           if ptp.message_type.general.dscp is defined and ptp.message_type.general.dscp is not none %}
+| Msg General | DSCP {{ ptp.message_type.general.dscp }} |
+{%             endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2:27]: indented more than matching `if`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+{%   if ptp.message_type is defined and ptp.message_type is not none %}
+{%       if ptp.message_type.general is defined and ptp.message_type.general is not none %}
+{%           if ptp.message_type.general.dscp is defined and ptp.message_type.general.dscp is not none %}
+ptp message-type general dscp {{ ptp.message_type.general.dscp }}
+{%             endif %}
+{%         endif %}
+{%       if ptp.message_type.event is defined and ptp.message_type.event is not none %}
+{%           if ptp.message_type.event.dscp is defined and ptp.message_type.event.dscp is not none %}
+ptp message-type event dscp {{ ptp.message_type.event.dscp }}
+{%             endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2:30]: indented more than matching `if`

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Small formatting changes required for readability, but functionality looks good.
Would be good to add more ptp interface knobs at some point (could just submit issue)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 (hunk 1 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2:5]: Fixed, thanks man!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2 (hunk 2 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2]: Fixed, thanks man!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2 (hunk 3 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ptp.j2:27]: Fixed, thanks man!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2 (hunk 4 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ptp.j2:30]: Fixed, thanks man!

**Comment by onurgashi** [comment]
Fixed em.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM - tested with avd-validation lab.

**Comment by sundayN** [comment]
This is great news. Looks like Christmas has come early!
Now we just need to add some more options to `ethernet_interfaces`
```
ptp:
  announce:
    interval: < integer>
  sync-message:
    interval: <-integer>
  delay-req: <-integer>
  role: < master | slave >
```

**Pull request merged**